lib/packwerk/commands/detect_stale_violations_command.rb

```diff
@@ -10,54 +10,56 @@
 require "packwerk/commands/offense_progress_marker"
 
 module Packwerk
-  class DetectStaleViolationsCommand
-    extend T::Sig
-    include OffenseProgressMarker
-    Result = Struct.new(:message, :status)
-
-    def initialize(files:, configuration:, run_context: nil, progress_formatter: nil, reference_lister: nil)
-      @configuration = configuration
-      @run_context = run_context
-      @reference_lister = reference_lister
-      @progress_formatter = progress_formatter
-      @files = files
-    end
+  module Commands
+    class DetectStaleViolationsCommand
+      extend T::Sig
+      include OffenseProgressMarker
+      Result = Struct.new(:message, :status)
+
+      def initialize(files:, configuration:, run_context: nil, progress_formatter: nil, reference_lister: nil)
+        @configuration = configuration
+        @run_context = run_context
+        @reference_lister = reference_lister
+        @progress_formatter = progress_formatter
+        @files = files
+      end
 
-    sig { returns(Result) }
-    def run
-      @progress_formatter.started(@files)
+      sig { returns(Result) }
+      def run
+        @progress_formatter.started(@files)
 
-      all_offenses = T.let([], T.untyped)
-      execution_time = Benchmark.realtime do
-        all_offenses = @files.flat_map do |path|
-          run_context.process_file(file: path).tap do |offenses|
-            mark_progress(offenses: offenses, progress_formatter: @progress_formatter)
+        all_offenses = T.let([], T.untyped)
+        execution_time = Benchmark.realtime do
+          all_offenses = @files.flat_map do |path|
+            run_context.process_file(file: path).tap do |offenses|
+              mark_progress(offenses: offenses, progress_formatter: @progress_formatter)
+            end
           end
         end
-      end
 
-      @progress_formatter.finished(execution_time)
-      calculate_result
-    end
+        @progress_formatter.finished(execution_time)
+        calculate_result
+      end
 
-    private
+      private
 
-    def run_context
-      @run_context ||= Packwerk::RunContext.from_configuration(@configuration, reference_lister: reference_lister)
-    end
+      def run_context
```

Comment:
I think the command class doesn't need to know how the run_context is defined. I think it would be better to keep it in the cli class.

Reply:
I'm not following how moving the `run_context` outside the command class will also remove the `reference_lister` dependency. It's being used to calculate the result here:
https://github.com/Shopify/packwerk/pull/78/files#diff-66788d8e2cafe199807e9963fd538243bcdd6ab0ea3203f07464cb93109e78c0R56
and

The options I see are to either pass also the `reference_lister` as a param, or to do `run_context.reference_lister. stale_violations?` to calculate the result but none of them convince me.. thoughts?

Reply:
Hmmm, I think the issue is that the `RunContext` class isn't a great abstraction for what we use it. But that is irrelevant to this PR. Let's keep the code you wrote 👍

Reply:
Once we have all the commands in their own classes it might be a good idea to try to refactor the RunContext 👌